Chatkit/Auth/TokenProvider.swift

```diff
@@ -0,0 +1,137 @@
+//
+//  TokenProvider.swift
+//  Chatkit
+//
+//  Created by Mike Pye on 06/12/2019.
+//  Copyright © 2019 Pusher Ltd. All rights reserved.
+//
+// STUB FILE. This file demonstrates the desired interface, but contains to implementation
+// TODO: Move many of these types to PusherPlatform
+
+import Foundation
+import PusherPlatform
+
+// TODO: Temporary
+extension String: Error { }
+
+// Re-export so that PusherPlatform import is not required in client code
+public typealias TokenProvider = PPTokenProvider
+public typealias TokenProviderResult = PPTokenProviderResult
+
+/// A factory for `TokenProvider`s.
```

Comment:
If first in the docs we explain that `ChatkitTokenProviders` is a factory wouldn't it be better to call it `ChatkitTokenProviderFactory`? Now it may sound a little bit more like it is a collection of ready to use (already instantiated) providers.

Also shouldn't `TokenProvider` be consistently called `ChatkitTokenProvider`? Also to better convey the information and avoid potential name clashes.

Reply:
I'm not averse to the name, but lots of people have a very unreasonable reaction to the word "factory" :/

Reply:
As long as we won't have factories of factories or more factories that actual types we shall be safe ;)